app/src/pages/inside/profilePage/modals/deleteAccountFeedbackModal/deleteAccountFeedbackModal.jsx

```diff
@@ -0,0 +1,151 @@
+    id: 'DeleteAccountFeedbackModal.NO_NEEDED',
+    defaultMessage: 'The account is no longer needed',
+  },
+  DISSATISFIED: {
+    id: 'DeleteAccountFeedbackModal.DISSATISFIED',
+    defaultMessage: 'Dissatisfied with the service',
+  },
+  ALTERNATIVE: {
+    id: 'DeleteAccountFeedbackModal.ALTERNATIVE',
+    defaultMessage: 'Found a better alternative',
+  },
+  OTHER: {
+    id: 'DeleteAccountFeedbackModal.OTHER',
+    defaultMessage: 'Other',
+  },
+});
+
+const NO_NEEDED = 'NO_NEEDED';
+const DISSATISFIED = 'DISSATISFIED';
+const ALTERNATIVE = 'ALTERNATIVE';
+const OTHER = 'OTHER';
+const OTHER_VALUE = 'OTHER_VALUE';
+
+const DeleteAccountFeedback = ({ invalid, handleSubmit }) => {
+  const { formatMessage } = useIntl();
+  const dispatch = useDispatch();
+
+  const continueButton = {
+    text: formatMessage(messages.continue),
+    onClick: () => {
+      handleSubmit((data) => {
+        dispatch(
+          showModalAction({
+            id: 'deleteAccountModal',
+            data: { data },
+          }),
+        );
+      })();
+    },
+    disabled: invalid,
+    attributes: { type: 'submit' },
+  };
+  const cancelButton = {
+    text: formatMessage(COMMON_LOCALE_KEYS.CANCEL),
+  };
+
+  return (
+    <ModalLayout
+      title={formatMessage(messages.header)}
+      okButton={continueButton}
+      cancelButton={cancelButton}
+    >
+      <div className={cx('description')}>{formatMessage(messages.description)}</div>
+      <form>
+        <ul className={cx('options')}>
+          {[NO_NEEDED, DISSATISFIED, ALTERNATIVE, OTHER].map((variant) => (
+            <li key={variant} className={cx('option')}>
+              <div className={cx('input')}>
+                <FieldProvider name={variant} format={Boolean}>
+                  <InputCheckbox>
+                    <div className={cx('label')}>{formatMessage(messages[variant])}</div>
+                  </InputCheckbox>
+                </FieldProvider>
+              </div>
+              {variant === OTHER && (
+                <div className={cx('input')}>
+                  <FieldProvider name={OTHER_VALUE}>
+                    <FieldErrorHint>
+                      <Input />
+                    </FieldErrorHint>
+                  </FieldProvider>
+                </div>
+              )}
+            </li>
+          ))}
+        </ul>
+      </form>
+    </ModalLayout>
+  );
+};
+DeleteAccountFeedback.propTypes = {
+  invalid: PropTypes.bool.isRequired,
+  handleSubmit: PropTypes.func.isRequired,
+};
+
+export const DeleteAccountFeedbackModal = withModal('deleteAccountFeedbackModal')(
+  reduxForm({
+    form: 'deleteAccountFeedbackForm',
+    validate: ({ OTHER_VALUE: value, ...checkBoxes }) => ({
+      [OTHER_VALUE]: bindMessageToValidator(
+        deleteAccountFeedbackOtherValue,
+        'textMore128Hint',
+      )(value),
+      [NO_NEEDED]: deleteAccountFeedbackAnyCheckboxIsChecked(checkBoxes),
+    }),
+    onChange: (data, dispatch, { change }) => {
```

Comment:
Can you dispatch the action conditionally here?
We don't need to dispatch it in case `OTHER` option already checked.

```javascript
const isOtherOptionTextExists = !!data[OTHER_VALUE];
if (!data[OTHER] && isOtherOptionTextExists) {
   dispatch(change(OTHER, isOtherOptionTextExists))
}
```

Reply:
Fixed.